pkg/providers/gitlab_client_test.go

```diff
@@ -0,0 +1,232 @@
+package providers
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/xanzy/go-gitlab"
+)
+
+func TestNewClientSucessful(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	MockGitlabClient := NewMockClientInterface(mockCtrl)
+	client := &Gitlab{
+		Token:   "example_token",
+		BaseURL: "example_url",
+		Api:     MockGitlabClient,
+	}
+	MockGitlabClient.EXPECT().NewClient(client.Token, client.BaseURL).Times(1)
+	client.Api.NewClient(client.Token, client.BaseURL)
+}
+
+func TestInitSucessful(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	MockGitlabClient := NewMockClientInterface(mockCtrl)
+	client := &Gitlab{
+		Token:   "Token",
+		BaseURL: "BaseURL",
+		Api:     MockGitlabClient,
+	}
+	MockGitlabClient.EXPECT().NewClient(client.Token, client.BaseURL).Times(1)
+	assert.Equal(t, nil, client.Init())
+}
+func TestInitMissingToken(t *testing.T) {
+	client := &Gitlab{
+		BaseURL: "BaseURL",
+	}
+	assert.Error(t, client.Init(), "Token can't be empty")
+}
+func TestEmptyBaseURL(t *testing.T) {
+	client := &Gitlab{
+		Token: "token",
+	}
+	assert.Equal(t, nil, client.Init())
+	assert.Equal(t, "https://gitlab.com/api/v4", client.BaseURL)
+}
+
+func TestListUsersSuccess(t *testing.T) {
```

Comment:
I'm not sure if changing the behavior of the ListUsers function wouldn't break any future tests.
Imagine that you call ListUsers("felipe") and the function returns "felipe", "marcelo", "douglas", I think this can cause some problems in the future

Reply:
Hm... If you're changing the behavior of that function then you should update the tests as well.

We could add the username in the ListUsers test struct as well, since it should have all "configurable" parameters to the test.